mypy/fastparse.py

```diff
@@ -455,7 +455,7 @@ def fix_function_overloads(self, stmts: List[Statement]) -> List[Statement]:
                 elif len(current_overload) > 1:
                     ret.append(OverloadedFuncDef(current_overload))
 
-                if isinstance(stmt, Decorator):
+                if isinstance(stmt, Decorator) and stmt.name != "_":
```

Comment:
What is this needed for?
(We probably need to do it in fastparse2 also)

Reply:
If there are multiple decorated functions right next to each other all named `_`, the parser creates an `OverloadedFuncDef` instead of using regular `FuncDef`s, and this change means we can just check for redefined `FuncDef`s instead of worrying about `OverloadedFuncDef`s.

Now that I think about it, it would probably be better to handle `OverloadedFuncDef`s that are named `_` in the semantic analyzer, instead of doing this.

Reply:
>Now that I think about it, it would probably be better to handle OverloadedFuncDefs that are named _ in the semantic analyzer, instead of doing this.

I did this in 385554f557345b8b117c868e5e7fa574c793e687. Could you take a look?